src/trees/MultiResolutionAnalysis.cpp

```diff
@@ -32,6 +32,22 @@
 
 namespace mrcpp {
 
+/** @returns New MultiResolutionAnalysis (MRA) object
+ *
+ *  @brief Contructs a MultiResolutionAnalysis object composed of computational domain (world) and a polynomial basis (Multiwavelets)
+ *
+ *  @param[in] bb: 2-element integer array [Lower, Upper] defining the bounds for a BoundingBox object representing the computational domain
+ *  @param[in] order: (integer) Maximum polynomial order of the multiwavelet basis, 
+ *  immediately used in the constructor of an InterPolatingBasis object which becomes an attribute of the MRA
+ *  @param[in] maxDepth: (integer) Exponent of the node refinement in base 2, relative to root scale. 
+ *  In other words, it is the maximum amount of refinement that we allow in a node, in other to avoid overflow of values.
+ *
+ *  @details Contructor of the MultiResolutionAnalysis class from scratch, without requiring any pre-existing complex structure. 
+ *  The constructor calls the InterpolatingBasis basis constructor to generate the MultiWavelets basis of functions,
+ *  then the BoundingBox constructor to create the computational domain. The constructor then checks if the generated node depth, or
+ *  node refinement is beyond the root scale or the maximum depth allowed, in which case it will abort the process.
+ *  Otherwise, the process goes on to setup the filters with the class' setupFilter method.
```

Comment:
```suggestion
 *  @param[in] bb: 2-element integer array [Lower, Upper] defining the bounds for a BoundingBox object representing the computational domain
 *  @param[in] order: Maximum polynomial order of the multiwavelet basis, 
 *  immediately used in the constructor of an InterPolatingBasis object which becomes an attribute of the MRA
 *  @param[in] maxDepth: Exponent of the node refinement in base 2, relative to root scale. 
 *  In other words, it is the maximum amount of refinement that we allow in a node, in other to avoid overflow of values.
 *
 *  @details Constructor of the MultiResolutionAnalysis class from scratch, without requiring any pre-existing complex structure. 
 *  The constructor calls the InterpolatingBasis basis constructor to generate the MultiWavelets basis of functions,
 *  then the BoundingBox constructor to create the computational domain. The constructor then checks if the generated node depth, or
 *  node refinement is beyond the root scale or the maximum depth allowed, in which case it will abort the process.
 *  Otherwise, the process goes on to setup the filters with the class' setupFilter method.
```
the type annotation is not needed because the documentation will be rendered _with_ the function signature and including the types.